pkg/controller/master/mcmsettings/serverurl.go

```diff
@@ -0,0 +1,38 @@
+package mcmsettings
+
+import (
+	"fmt"
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// updateServerURL will reconcile ingress-expose service in kube-system namespace and set the
+// external IP to rancher server-url setting
+func (h *mcmSettingsHandler) updateServerURL(key string, svc *corev1.Service) (*corev1.Service, error) {
+	if svc == nil || svc.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if key != vipService {
+		return svc, nil
+	}
+
+	if len(svc.Status.LoadBalancer.Ingress) != 1 {
+		return svc, fmt.Errorf("expected to find 1 load balancer ip for svc %s but got %v", key, svc.Spec.ExternalIPs)
+	}
+
+	serverURLSetting, err := h.rancherSettingsCache.Get(rancherServerURLSetting)
+	if err != nil {
+		return svc, fmt.Errorf("error querying rancher server-url setting: %v", err)
+	}
+
+	serverURLSettingCopy := serverURLSetting.DeepCopy()
```

Comment:
I believe we only need to configure it once, future updates should be fully controled by the users.

Reply:
this is now fixed, and we will only sync the setting the first time. 
any subsequent changes will be ignored.